src/kontrol/foundry.py

```diff
@@ -268,12 +268,19 @@ def method_digest(self, contract_name: str, method_sig: str) -> str:
     def contract_name_from_bytecode(self, bytecode: bytes) -> str:
         for contract_name, contract_obj in self.contracts.items():
             zeroed_bytecode = bytearray(bytecode)
-            for start, length in contract_obj.immutable_ranges:
+            deployed_bytecode_str = re.sub(
+                pattern='__\\$(.){34}\\$__',
+                repl='0000000000000000000000000000000000000000',
+                string=contract_obj.deployed_bytecode,
+            )
+            deployed_bytecode = bytearray.fromhex(deployed_bytecode_str)
```

Comment:
It would be nice to have a unit test for this function in [src/tests/unit/test_solc_to_k.py](https://github.com/runtimeverification/kontrol/blob/master/src/tests/unit/test_solc_to_k.py)

Reply:
Added.